File: app/src/main/java/mishka/rssreader/ui/feed/FeedViewModel.java
package mishka.rssreader.ui.feed;

import android.arch.lifecycle.ViewModel;

import javax.inject.Inject;

import io.realm.RealmResults;
import mishka.rssreader.data.DataManager;
import mishka.rssreader.data.model.RealmRssItem;

public class FeedViewModel extends ViewModel {
    private final RealmResults<RealmRssItem> realmResults;
    private DataManager dataManager;

    @Inject
    public FeedViewModel(DataManager dataManager) {
        this.dataManager = dataManager;
        realmResults = dataManager.getRssItems();
    }

    public void update() {
        dataManager.update();
    }

    public RealmResults<RealmRssItem> getRealmItems() {
        return realmResults;
    }
}

File: app/src/main/java/mishka/rssreader/ui/feed/FeedViewModel.kt
package mishka.rssreader.ui.feed;

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.ViewModel


import javax.inject.Inject

import mishka.rssreader.data.DataManager
import mishka.rssreader.data.model.RssItem

class FeedViewModel @Inject constructor (val dataManager: DataManager): ViewModel() {
    val posts: LiveData<List<RssItem>> = dataManager.getPosts()

    fun update() = dataManager.update()
}
